src/execution_plan/record.c

```diff
@@ -15,12 +15,15 @@ static void _RecordPropagateEntry
 (
 	Record dest,
 	Record src,
-	uint idx
+	uint idx,
+	bool make_volatile
 ) {
 	Entry e = src->entries[idx];
 	dest->entries[idx] = e;
 	// if the entry is a scalar, make sure both Records don't believe they own the allocation
-	if(e.type == REC_TYPE_SCALAR) SIValue_MakeVolatile(&src->entries[idx].value.s);
+	if(make_volatile && e.type == REC_TYPE_SCALAR) {
+		SIValue_MakeVolatile(&src->entries[idx].value.s);
+	}
```

Comment:
The comment above: "make sure both Records don't believe they own the allocation" is misleading
If I understand correctly only the `src` ownership is changed, the `dest` record is the new owner of the allocation.

Reply:
Also I think `make_volatile` can get us into trouble assuming `make_volatile` is `false` then both `src` and `dest` assume ownership over the allocation.